src/Uno.Analyzers/UnoDoNotDisposeNativeViews.cs

```diff
@@ -0,0 +1,148 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Uno.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public class UnoDoNotDisposeNativeViews : DiagnosticAnalyzer
+	{
+		internal const string Title = "Do not dispose native views";
+		internal const string MessageFormat = "Do not call Dispose() on {0}";
+		internal const string Description = "Calling Dispose on native controls can cause undefined behaviors and runtime crashes.";
+		internal const string Category = "Reliability";
+
+		internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+#pragma warning disable RS2008 // Enable analyzer release tracking
+			"Uno0002",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+			Title,
+			MessageFormat,
+			Category,
+			DiagnosticSeverity.Warning,
+			isEnabledByDefault: true,
+			description: Description,
+			helpLinkUri: "https://aka.platform.uno/UNO0002"
+		);
+
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+			context.RegisterCompilationStartAction(context =>
+			{
+				var uiViewSymbol = context.Compilation.GetTypeByMetadataName("UIKit.UIView");
+				if (uiViewSymbol is null)
+				{
+					return;
+				}
+
+				context.RegisterOperationAction(
+					c => AnalyzeOperation(c, uiViewSymbol)
+					, OperationKind.Invocation
+					, OperationKind.Using
+					, OperationKind.UsingDeclaration
+					, OperationKind.DeclarationPattern);
+			});
+		}
+
+		private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol uiViewSymbol)
+		{
+			if (context.Operation is IInvocationOperation invocationOperation)
+			{
+				if (invocationOperation.TargetMethod.Name == "Dispose"
+					&& IsType(invocationOperation.TargetMethod.ContainingType, uiViewSymbol)
+					&& (
+						invocationOperation.TargetMethod.Parameters.Length == 0
+						|| (
+							invocationOperation.TargetMethod.Parameters.Length == 1
+							&& invocationOperation.TargetMethod.Parameters[0].Type.SpecialType == SpecialType.System_Boolean)))
+				{
+					var diagnostic = Diagnostic.Create(
+						Rule,
+						context.Operation.Syntax.GetLocation(),
+						invocationOperation.TargetMethod.ToDisplayString()
```

Comment:
Would the error message say "Don't call Dispose() on UIKit.UIView.Dispose()"? That sounds a bit strange.

Reply:
True, will adjust!